internal/controller/model_nodeport.go

```diff
@@ -40,27 +48,51 @@ func NewNodePortLoadBalancerModelGenerator(
 	}
 }
 
-func (g *NodePortLoadBalancerModelGenerator) getDestinationAddresses() (result []string, err error) {
+func isValidIp(ipString string) (err error) {
+	ipParsed := net.ParseIP(ipString)
+	if ipParsed != nil {
+		return nil
+	}
+	return ErrInvalidIpAddress
+}
+
+func isIPv4Address(ipString string) (isIPv4 bool) {
+	return strings.Count(ipString, ":") == 0 && strings.Count(ipString, ".") == 3
```

Comment:
Fun fact: The following are valid IP addresses, at least on most UNIX systems:

- `0` (equivalent to `0.0.0.0`)
- `127.1` (equivalent to `127.0.0.1`)
- `10.133989` (equivalent to `10.2.11.101`)

But I don’t think we should/need to support those use cases :)